src/pybind/ceph_volume_client.py

```diff
@@ -439,19 +441,33 @@ def create_volume(self, volume_path, size=None, data_isolated=False):
         path = self._get_path(volume_path)
 
         self._mkdir_p(path)
-
         if size is not None:
             self.fs.setxattr(path, 'ceph.quota.max_bytes', size.__str__(), 0)
 
+        done = False;
         if data_isolated:
-            pool_name = "{0}{1}".format(self.POOL_PREFIX, volume_path.volume_id)
-            pool_id = self._create_volume_pool(pool_name)
-            mds_map = self._rados_command("mds dump", {})
-            if pool_id not in mds_map['data_pools']:
-                self._rados_command("mds add_data_pool", {
-                    'pool': pool_name
-                })
-            self.fs.setxattr(path, 'ceph.dir.layout.pool', pool_name, 0)
+            isolate_by_namespace = bool(self.rados.conf_get("cephFSClient_data_isolate_by_namespace"))
+            if isolate_by_namespace:
+                namespace = "{0}{1}".format(self.POOL_NS_PREFIX, volume_path.volume_id)
+                log.info("create_volume: {0}, using rados namespace {1} to isolate data.".format(volume_path, namespace))
+                try:
+                    self.fs.setxattr(path, 'ceph.dir.layout.pool_namespace', namespace, 0)
+                except:
```

Comment:
All exception handlers must specify what exception they are catching, catch-all handlers are bad because they might catch things like connection errors (we should let these get raised further), rather than the kind of thing we are expecting (errors from the particular filesystem call)